refinery/data_set_manager/utils.py

```diff
@@ -1192,3 +1194,94 @@ def get_solr_response_json(node_uuids):
+    :param existing_dataset_uuid: the UUID of an existing DataSet
+    """
+    existing_dataset = core.models.DataSet.objects.get(
+        uuid=existing_dataset_uuid
+    )
+    associate_datafiles_from_existing_dataset(
+        existing_dataset, revised_investigation
+    )
+    updated_dataset_title = revised_investigation.title
+    existing_dataset.update_investigation(
+        revised_investigation,
+        "Metadata Revision: for {}".format(updated_dataset_title)
+    )
+    data_set_manager.tasks.annotate_nodes(revised_investigation.uuid)
+    existing_dataset.set_title(updated_dataset_title)
+
+
+@transaction.atomic()
+def associate_datafiles_from_existing_dataset(existing_dataset,
+                                              revised_investigation):
+    """
+    Transfer data files from an existing DataSet's FileStoreItems that
+    correspond to data files of the same source in an Investigation's
+    FileStoreItems
+    :param existing_dataset: DataSet instance to transfer datafiles from
+    :param revised_investigation: Investigation instance to transfer data files
+     to
+    """
+    _check_data_set_cleanliness(existing_dataset)
+    _ensure_revised_investigation_references_existing_datafiles(
+        existing_dataset, revised_investigation
+    )
+
+    existing_investigation = existing_dataset.get_investigation()
+    file_store_items_from_existing_dataset = \
+        existing_investigation.get_file_store_items(local_only=True)
+
+    for new_file_store_item in revised_investigation.get_file_store_items():
+        for prior_file_store_item in file_store_items_from_existing_dataset:
+            if prior_file_store_item.source == new_file_store_item.source:
+                new_file_store_item.datafile = prior_file_store_item.datafile
+                new_file_store_item.save()
+                # It's crucial to clear the datafile of the prior
+                # FileStoreItem as well. Otherwise there would be two
+                # references to the same data file which could cause
+                # unintended side-effects
+                prior_file_store_item.datafile = None
+                prior_file_store_item.save()
+
+
+def _check_data_set_cleanliness(data_set):
+    if not data_set.is_clean():
+        raise RuntimeError("DataSet with UUID: {} is not clean (There have "
+                           "been Analyses or Visualizations performed on it) "
+                           "Remove these objects and try again"
+                           .format(data_set.uuid))
+
+
+def _ensure_revised_investigation_references_existing_datafiles(
+        existing_dataset, revised_investigation
+):
+    existing_investigation = existing_dataset.get_investigation()
+    existing_datafile_names, new_datafile_names = (
+        investigation.get_datafile_names(exclude_metadata_file=True) for
+        investigation in [existing_investigation, revised_investigation]
+    )
```

Comment:
plain old assignment might be fine here, and/or, make both sets now, since that's what you need?